extensions/binding/org.eclipse.smarthome.binding.bosesoundtouch/src/main/java/org/eclipse/smarthome/binding/bosesoundtouch/internal/discovery/SoundTouchDiscoveryParticipant.java

```diff
@@ -52,20 +53,35 @@
     }
 
     @Override
+    @SuppressWarnings("deprecation")
     public DiscoveryResult createResult(ServiceInfo info) {
         DiscoveryResult result = null;
         ThingUID uid = getThingUID(info);
         if (uid != null) {
 
+            // remove the domain from the name
+            InetAddress[] addrs = info.getInetAddresses();
+
             Map<String, Object> properties = new HashMap<>(2);
             String label = "Bose SoundTouch";
             try {
-                label = info.getName();
+                if (BST_10_THING_TYPE_UID.equals(uid.getThingTypeUID())) {
+                    try {
+                        String group = DiscoveryUtil
+                                .executeUrl("http://" + addrs[0].getHostAddress() + ":8090/getGroup");
+                        label = DiscoveryUtil.getContentOfFirstElement(group, "name");
+                    } catch (Exception e) {

```

Comment:
using try/catch for process flow is not nice. Can you check against some different condition? At least please only catch known exceptions (IOException here).